samples/kernel/metairq_dispatch/src/main.c

```diff
@@ -120,13 +118,6 @@ static void record_latencies(struct msg *m, uint32_t latency)
 		return;
 	}
 
-	k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
I'm confused here.  I'm just now noticing the earlier patch, but... it looks correct to me.  This is a test of preemptible thread behavior, so locking of some form between the threads is probably what we want.

Reply:
But to be clear: it doesn't have to be a spinlock.  There's no ISR context and (as you point out) we're running with MP_NUM_CPUS=1, so anything will do: a k_mutex() or k_sem() would be fine, and (as long as we're never SMP) so would sched_un/lock().

Reply:
Yes, @andyross , in previous patch I did not aware of this is a 1 CPU scenario, like you pointing out, it only does irq_lock here, and no ISR context. (Actually I tried it doesn't work at all to fix this issue).  So the spinlock should be removed here.

And I try on another way to fix this, by just checking if the message sequence number > MAX_EVENT) when entering the record_latencies subroutine  (in previously I check the stats.mirq but it still failed sometimes due to this shared variable can also have a race. 

Look like it works well and no need to protect a section of code. And I tested it on both ehl_crb and acrn_ehl_crb then the test pass more than 20 times. Actually this failure can be reproduced within 5~8 times previously.

Do you think is this solution ok? Thank you!